src/richie/apps/search/filter_definitions/courses.py

```diff
@@ -74,9 +74,20 @@ def get_aggs_fragment(self, queries, data, *args, **kwargs):
         Build the aggregations as a term query that counts all the different values assigned
         to the field.
         """
-        # Look for an include parameter in the form data and default to the regex returned
-        # by the `aggs_include` property:
-        include = data[f"{self.name:s}_include"] or self.aggs_include
+        # Look the aggregations parameters in the form data (either [filter]_children_aggs or
+        # [filter]_aggs), and default to the filter definition's aggs_include.
+        if data[f"{self.name:s}_children_aggs"]:
+            page = Page.objects.get(
+                node__path=data[f"{self.name:s}_children_aggs"][2:],
+                publisher_is_draft=False,
+            )
+            paths = page.node.get_children().values_list("path", flat=True)
+            # TO DO: replace this with proper IDs
+            # In the meantime, this should add all necessary categories to aggregations
+            # without making additional requests.
+            include = [f"{prefix}-{path}" for path in paths for prefix in ["P", "L"]]
```

Comment:
Should we cache it for a minute in memory for scaling since it is inducing a few db queries?